### ISSUE TITLE
Copying database with view alters view syntax

### ISSUE THREAD

**Issue by dillyhammer** [phpmyadmin/phpmyadmin#12818]
### Steps to reproduce
1. Create a database with table(s) to be used in a view
2. Create a view that references the table(s)
3. Copy database to a new database with a new name

### Expected behaviour
The new database should have a view with the exact same definition as used to create it in the original database, without any changes whatsoever.

### Actual behaviour
The syntax used to create the view in the new database has been altered.
Examples include:
1. References to tables in the view include backticked references to the original database. So the view on the new database can contain a query such as "select * from `old_database`.`table_name`". 
2. A query written as "where column_name != 1" is changed to "where column_name <>1" - while this may appear to be the same it is not, subtle as it is.
3. There may be other changes, these two are what struck me as dangerous. At this time, migrating views from a master database back to 6 others requires manually editing ALL copied views to reference proper databases.

### Server configuration
**Operating system**:
CentOS Linux 7.2.1511

**Web server:**
Apache version 2.4.6

**Database:**
Database server version: 5.5.50-MariaDB
Database client version: libmysql - mysqlnd 5.0.11-dev - 20120503

**PHP version:**
php55-php 5.5.21-5.el7

**phpMyAdmin version:**
4.6.5.2

### Client configuration
**Browser:**
Chrome Version 55.0.2883.87 m

**Operating system:**
Win7

**Comment by nijel** [comment]
I'm afraid phpMyAdmin is not guilty here, all we do is `SHOW CREATE TABLE` and use this output. Can you please check what it returns for your views?

**Comment by dillyhammer** [comment]
Using the SHOW CREATE TABLE for the original view at the mysql shell returns the proper definition for the view:

USE `database_name1`;

SHOW CREATE TABLE `view_name`
Output is : SELECT * FROM `table_name`

I then use phpmyadmin to copy the database `database_name1` to `database_name2`;
Again, at the shell:

USE `database_name2`;
SHOW CREATE TABLE `view_name`
Output is : SELECT * FROM `database_name1`.`table_name`

Using phpmyadmin on `database_name1`, I select  `view_name`, click "Structure" and click "Edit View":
Output is : SELECT * FROM `database_name1`.`table_name`
This is unexpected behaviour. The database name was not used to create the view.
This output is different than that at the shell.

Using phpmyadmin on `database_name2`, I select  `view_name`, click "Structure" and click "Edit View":
Output is : SELECT * FROM `database_name1`.`table_name`

Using phpmyadmin on `database_name1`, if I use the export feature to display the structure of the view, I get the correct behaviour:
Output is : SELECT * FROM `table_name`

Using phpmyadmin on `database_name2`, if I use the export feature to display the structure of the view, I get unpected behaviour:
Output is : SELECT * FROM `database_name1`.`table_name`

When I use phpmyadmin to copy databases containing views, I have to go back to each and every view and remove the database name from the definition.

This is very reproduceable.

**Comment by ibennetch** [comment]
I've also done a number of tests and it seems phpMyAdmin is consistent with MySQL here, even though it's annoying and not ideal. MySQL is adding database and table names to every view I create, even when I leave them out of my definition.

In fact, I haven't been able to get MySQL to create a view where it doesn't add the database and table names in this way.

**Comment by dillyhammer** [comment]
Well, I have not been able to get MySQL to create a view where it - of it's own accord - adds the database name to the table name.  In my tests, using the software as set out in my report, only phpmyadmin is doing this. It's corrupting my databases. Well beyond annoying.

If you folks aren't able to reproduce this, for whatever reason, not much I can do about that.

Out.

**Comment by ibennetch** [comment]
I wonder if this is a difference between MariaDB and MySQL behaviors. Maybe someone else with MariaDB can do some more tests; I don't have it installed at the moment.

**Comment by dillyhammer** [comment]
One last thing. I switched to MySQL Workbench out of curiosity. The tool does not exhibit any of these issues. It displays, exports, alters and copies views precisely as originally defined. I tested this on both my MariaDB and MySQL Servers. Both showed the exact same expected behaviour.

**Comment by ibennetch** [comment]
I've downloaded Workbench to try to reproduce this and I'm seeing the same behavior there on my MySQL server.

There's a close match in a [bug report](https://bugs.mysql.com/bug.php?id=40345) at MySQL for version 5.1, but no reports about any version near yours. 

http://stackoverflow.com/questions/28676723/mysql-create-view-across-all-prefixed-databases-table makes it sound like this is standard behavior.

In summary, I'm not sure what to do next; I'm sympathetic to the problem you're having, but without being able to reproduce it I'm having trouble fixing it.

**Comment by dillyhammer** [comment]
Reading that bug report, yeah, that's the behaviour. It looks like it was fixed in MySQL > 5.0.79. Here's the thing. If you look at my report, my *server* version is "5.5.50-MariaDB", but my *client* version is "libmysql - mysqlnd 5.0.11-dev - 20120503".

IMHO, I think the problem here exists in the client, which is <  5.0.79. I have no other explanation, and it kind of makes sense if I'm unable to reproduce the issue at the 5.5.50 command line, but see it when using the web-based 5.0.11 client.

In short, I no longer believe this is a phpmyadmin issue, and the report can be safely closed. At least we've left behind some decent residue for others to refer to - surely I'm not the only developer bit on the tail by this.

Thanks fro your time on this, I appreciate it.
